.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -2106,7 +2106,19 @@ private DateTimeResolutionResult ParseWhichWeek(string text, DateObject referenc
                     return ret;
                 }
 
-                var year = referenceDate.Year;
+                var year = config.DateExtractor.GetYearFromText(match.Match);
+                if (year == Constants.InvalidYear)
+                {
+                    var orderStr = match.Groups["order"].Value;
```

Comment:
Could we have code comment here to exemplify what falls into this if?
Also, "order" should reuse a named group constant.